`make codegen` is no longer working

**Describe the bug**
Invoking the command will fail with:
```text
=== Start resetting changes introduced by YEAR ===
fatal: Unable to create '/go/src/github.com/vmware-tanzu/antrea/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
make: *** [codegen] Error 128
##[error]Makefile:250: recipe for target 'codegen' failed
##[error]Process completed with exit code 2.
```

This is causing the `Check code generation` job in the `Go` workflow to fail.

**To Reproduce**
Run `make codegen` locally.

**Additional context**
The following code:
https://github.com/vmware-tanzu/antrea/blob/c91d6ee5c6482dd9cf65f90adbba2494381c264b/hack/update-codegen-dockerized.sh#L148-L154

may run 2 git processes simultaneously (`git diff  --numstat` and `git diff <file>`). This becomes more likely as the list of files returned by `git diff  --numstat` is longer, because of how piping works. With the year change (2020 -> 2021), this list is now quite large. This issue can easily be solved by adding an invocation to the `tac` command before starting to iterate over the list of files.

--- a/hack/update-codegen-dockerized.sh
+++ b/hack/update-codegen-dockerized.sh
@@ -146,7 +146,10 @@ rm -rf /tmp/includes
 set +x
 
 echo "=== Start resetting changes introduced by YEAR ==="
-git diff  --numstat | awk '$1 == "1" && $2 == "1" {print $3}' | while read file; do
+# The call to 'tac' ensures that we cannot have concurrent git processes, by
+# waiting for the call to 'git diff  --numstat' to complete before iterating
+# over the files and calling 'git diff ${file}'.
+git diff  --numstat | awk '$1 == "1" && $2 == "1" {print $3}' | tac | while read file; do
   if [[ "$(git diff ${file})" == *"-// Copyright "*" Antrea Authors"* ]]; then
     git checkout HEAD -- "${file}"
     echo "=== ${file} is reset ==="

